src/interpreter/virtualMachine.ts

```diff
@@ -0,0 +1,295 @@
+import {
```

Comment:
Putting this comment here since there is no way to directly comment on the file name. Wouldn't the virtual machine be the evaluator instead? From notes 4:
> To this aim, we are going to translate Source to a sequence of instructions prior to execution. We then hand over the instruction sequence, which is called virtual machine code, to a virtual machine.